libsnark/gadgetlib1/pb_variable.hpp

```diff
@@ -163,6 +163,17 @@ class pb_linear_combination_array
     FieldT get_field_element_from_bits(const protoboard<FieldT> &pb) const;
 };
 
+/// A utility function which creates and allocates a variable in a single step
+/// (and can therefore be used in initalizer lists, which greatly simplifies
+/// many constructors).
+///
+/// TODO: Why does pb_variable not have an allocating constructor of this form,
+/// even further simplifying a lot of code. Move this to an appropriate
+/// constructor if there are no issues.
+template<typename FieldT>
+pb_variable<FieldT> pb_variable_allocate(
+    protoboard<FieldT> &pb, const std::string &annotation);
```

Comment:
I put it here since that seemed to be consistent with all other convenience functions being the bottom.
I don't have a particularly strong opinion about the location though, so happy to move it if you prefer.

Reply:
AFAICT, the convenience functions at the end of the file are all about `pb_linear_combination_array` which is the class declared just above. I'd move the `pb_variable_allocate` just below the relevant class to group by classes and related "utility functions".

Reply:
Moved function declaration.

Reply:
Why not move it straight under the `pb_variable` class (instead of moving it under `pb_variable_array`)?

Reply:
Oops.  Sorry.  Moved it again.

Reply:
Thanks